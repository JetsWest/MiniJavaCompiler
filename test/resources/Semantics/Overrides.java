class SomeClass {
    public static void main(String[] args) {
        System.out.println(0);
    }
}

class A {
    public int foo() {
        return 0;
    }
}

class B extends A {
    public int foo() {
        return 1;
    }
}

class C extends A {
    public boolean foo() {
        return true;
    }
}

class D extends A {
    public int foo(int x) {
        return 2;
    }
}

class E {
    public int foo(int x) {
        return 3;
    }
}

class F extends E {
    public int foo(boolean x) {
        return 0; // needs to be integer
    }
}