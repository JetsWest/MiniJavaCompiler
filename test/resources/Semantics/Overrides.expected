(Line 20) Overrode methods must have same return type.
(Line 26) Overrode methods must have same number of parameters.
(Line 38) Overrode methods must have same parameter types.
